<a href="https://colab.research.google.com/github/gemini106/KubernetesGoogle/blob/master/Movies.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import numpy as np 


In [ ]:
mv_data=pd.read_csv('movie.csv')

In [ ]:
mv_data.shape

(16744, 17)

# We have 16,744 rows and 17

In [ ]:
mv_data.head(20)

Unnamed: 0  ID  ...                                     Language  Runtime
0            0   1  ...                      English,Japanese,French    148.0
1            1   2  ...                                      English    136.0
2            2   3  ...                                      English    149.0
3            3   4  ...                                      English    116.0
4            4   5  ...                                      Italian    161.0
5            5   6  ...                              English,Spanish    117.0
6            6   7  ...                       English,German,Russian    150.0
7            7   8  ...                English,German,French,Italian    165.0
8            8   9  ...  English,German,Hebrew,Spanish,Arabic,Nepali    115.0
9            9  10  ...                English,German,French,Italian    153.0
10          10  11  ...                              English,Spanish    114.0
11          11  12  ...                                Hindi,English    170.0
12          12  13  ...                                      Spanish    118.0
13          13  14  ...                                      English    118.0
14          14  15  ...                         English,French,Latin     91.0
15          15  16  ...                      Italian,English,Spanish    165.0
16          16  17  ...                  English,German,Greek,Arabic    127.0
17          17  18  ...                       English,French,Italian    101.0
18          18  19  ...                                      English    118.0
19          19  20  ...                                      English    126.0

[20 rows x 17 columns]

Dropping Unammed as it holds no useful info, or anything we can extract from, basically its like dirt

Dropping ID, because it serves like Sr No only ( One can keep it if needed)

In [ ]:
#Remove "Unnamed" 
#mv_data=mv_data.drop("Unnamed: 0",axis=1)

In [ ]:
#Remove "Id" 
#mv_data=mv_data.drop("ID",axis=1)

In [ ]:
#Easier way to drop coloumns at once
mv_data = mv_data.drop(columns=['Unnamed: 0', 'ID'])

In [ ]:
mv_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16744 entries, 0 to 16743
Data columns (total 15 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Title            16744 non-null  object 
 1   Year             16744 non-null  int64  
 2   Age              7354 non-null   object 
 3   IMDb             16173 non-null  float64
 4   Rotten Tomatoes  5158 non-null   object 
 5   Netflix          16744 non-null  int64  
 6   Hulu             16744 non-null  int64  
 7   Prime Video      16744 non-null  int64  
 8   Disney+          16744 non-null  int64  
 9   Type             16744 non-null  int64  
 10  Directors        16018 non-null  object 
 11  Genres           16469 non-null  object 
 12  Country          16309 non-null  object 
 13  Language         16145 non-null  object 
 14  Runtime          16152 non-null  float64
dtypes: float64(2), int64(6), object(7)
memory usage: 1.9+ MB


In [ ]:
#You can Run seperate as well
mv_data.dtypes

Title               object
Year                 int64
Age                 object
IMDb               float64
Rotten Tomatoes     object
Netflix              int64
Hulu                 int64
Prime Video          int64
Disney+              int64
Type                 int64
Directors           object
Genres              object
Country             object
Language            object
Runtime            float64
dtype: object

In [ ]:
pd.isnull(mv_data).sum()

Title                  0
Year                   0
Age                 9390
IMDb                 571
Rotten Tomatoes    11586
Netflix                0
Hulu                   0
Prime Video            0
Disney+                0
Type                   0
Directors            726
Genres               275
Country              435
Language             599
Runtime              592
dtype: int64

# Basically all those with 0, have all unique value
 eg. 16744 titles equal to the number of rows in the dataset.So,there are no duplicate entries.

# Also there are no Null enteries in them 

# Age

Since more than 60% of the movies don`t have data for Age in the dataset, we will replace the NaN values with 'Unknown':

In [ ]:
mv_data.Age = mv_data.Age.fillna('Unknown')

In [ ]:
mv_data.Age.value_counts()

18+    3474
7+     1462
13+    1255
all     843
16+     320
Name: Age, dtype: int64

In [ ]:
pd.isnull(mv_data).sum()

Title                  0
Year                   0
Age                    0
IMDb                 571
Rotten Tomatoes    11586
Netflix                0
Hulu                   0
Prime Video            0
Disney+                0
Type                   0
Directors            726
Genres               275
Country              435
Language             599
Runtime              592
dtype: int64

IMDb and Rotten Tomatoes columns

We will also replacing empty values for the IMDb column. Rotten Tomatoes has pecentage signs on the data. We will strip it and convert the values to float for performing numerical calculations:

In [ ]:
mv_data.IMDb = mv_data.IMDb.fillna(0)


In [ ]:
#stripping the % sign and converting to float
mv_data['Rotten Tomatoes'] = mv_data['Rotten Tomatoes'].str.rstrip('%').astype('float')
#Filling NaN values with 0
mv_data['Rotten Tomatoes'] = mv_data['Rotten Tomatoes'].fillna(0)

Incase you want the prediction, based only on IMDB, drop Rotten Tomatoes coloumn, as it has excessively low data.

secondly, if you keep Rotten Tomatoes and fill empty ones with 0, automatically will increase the status of those movies which have good rating on IMDB as well a descent rating on Rotten Tomatoes

In [ ]:
#Dropping "Rotten Tomatoes" for its excessively low data 
#mv_data = mv_data.drop(['Rotten Tomatoes'], axis=1)

#"Title" and "Type" are dropped as it is irrelevant for our prediction.
#mv_data = mv_data.drop(['Title','Type'], axis=1)
#mv_data.isna().sum()

In [ ]:
pd.isnull(mv_data).sum()

Title                0
Year                 0
Age                  0
IMDb                 0
Rotten Tomatoes      0
Netflix              0
Hulu                 0
Prime Video          0
Disney+              0
Type                 0
Directors          726
Genres             275
Country            435
Language           599
Runtime            592
dtype: int64

In [ ]:
mv_data.describe()

Year          IMDb  ...     Type       Runtime
count  16744.000000  16744.000000  ...  16744.0  16152.000000
mean    2003.014035      5.701457  ...      0.0     93.413447
std       20.674321      1.703681  ...      0.0     28.219222
min     1902.000000      0.000000  ...      0.0      1.000000
25%     2000.000000      4.900000  ...      0.0     82.000000
50%     2012.000000      6.000000  ...      0.0     92.000000
75%     2016.000000      6.900000  ...      0.0    104.000000
max     2020.000000      9.300000  ...      0.0   1256.000000

[8 rows x 9 columns]

There seems to be a movie with a Runtime of 1256. Let's confirm if any mistakes have hapenned with this column:



In [ ]:
#Checking if there are movies with more than 300 minutes of runtime
mv_data[mv_data.Runtime > 300]

Title  ...  Runtime
4405     Law of the Lawless  ...    750.0
12475              Scarlett  ...    360.0
13048   Custer's Last Stand  ...    328.0
13115  Carlos el terrorista  ...    334.0
13179              Colorado  ...   1256.0
14113      The Inner Circle  ...    360.0
15295   The Vatican Museums  ...    410.0

[7 rows x 15 columns]

There really are some mistakes in the data. For example, "Colorado" shows as having 57 minutes of runtime in IMDb catalog, and "Law of the Lawless" has 87 minutes. Since it is more common to have movies that are three hours long (around 200min) and there are only 7 entries above 300min 

Thus we will correct these fields manually using the data available at www.imdb.com:



In [ ]:
mv_data.loc[mv_data.Title == 'Law of the Lawless', ['Runtime']] = 87
mv_data.loc[mv_data.Title == 'Scarlett', ['Runtime']] = 92
mv_data.loc[mv_data.Title == 'Carlos el terrorista', ['Runtime']] = 97
mv_data.loc[mv_data.Title == 'Colorado', ['Runtime']] = 57
mv_data.loc[mv_data.Title == 'The Inner Circle', ['Runtime']] = 96

Note: Custer's Last Stand and The Vatican Museums didn't need to be corrected. These two are really long!



In [ ]:
mv_data.describe()

Year          IMDb  ...     Type       Runtime
count  16744.000000  16744.000000  ...  16744.0  16152.000000
mean    2003.014035      5.701457  ...      0.0     93.250557
std       20.674321      1.703681  ...      0.0     25.954224
min     1902.000000      0.000000  ...      0.0      1.000000
25%     2000.000000      4.900000  ...      0.0     82.000000
50%     2012.000000      6.000000  ...      0.0     92.000000
75%     2016.000000      6.900000  ...      0.0    104.000000
max     2020.000000      9.300000  ...      0.0    410.000000

[8 rows x 9 columns]

In [ ]:
mv_data.update(mv_data[['Directors', 'Genres', 'Country', 'Language']].fillna('Unknown'))

In [ ]:
pd.isnull(mv_data).sum()

Unnamed: 0         0
ID                 0
Title              0
Year               0
Age                0
IMDb               0
Rotten Tomatoes    0
Netflix            0
Hulu               0
Prime Video        0
Disney+            0
Type               0
Directors          0
Genres             0
Country            0
Language           0
Runtime            0
dtype: int64